conan/tools/cmake/layout.py

```diff
@@ -51,8 +51,12 @@ def cmake_layout(conanfile, generator=None, src_folder=".", build_folder="build"
 
 def get_build_folder_custom_vars(conanfile):
 
-    build_vars = conanfile.conf.get("tools.cmake.cmake_layout:build_folder_vars",
-                                    default=[], check_type=list)
+    if conanfile.tested_reference_str:
```

Comment:
After your explanation I quite like this. My only gripe would be that as far as I can see, we have now lost the `test_output` bit in the path, so it'd be harder to gitignore these folders now.
May be worth re-adding them before merging (Or if it's still there, please point me to the code that does that as I must have missed it!)

Reply:
This is a ``cmake_layout``, the base folder name is already an argument to the function (default ``"build"``), so that will still be present, is the same default name that must be ignored for local builds (``conan install .``) too.